app/controllers/vm_common.rb

```diff
@@ -136,6 +136,10 @@ def websocket_use_ssl?
   end
   private :websocket_use_ssl?
 
+  def hide_vms
+    !User.current_user.settings.fetch_path(:display, :display_vms) # default value is false
```

Comment:
I find it quite confusing that the settings is called `display_vms`, but the method is called `hide_vms`.  I think it would be better to keep the names consistent.  I personally prefer to keep things in the positive, so would prefer `display_vms`, but that's up to you.  Thoughts?